pkg/config/config.go

```diff
@@ -102,18 +90,28 @@ type MicroshiftConfig struct {
 	BaseDomain       string        `json:"baseDomain"`
 	Cluster          ClusterConfig `json:"cluster"`
 
-	Ingress IngressConfig      `json:"-"`
-	Etcd    InternalEtcdConfig `json:"etcd"`
+	Ingress IngressConfig `json:"-"`
+	Etcd    EtcdConfig    `json:"etcd"`
 }
 
 // Top level config file
 type Config struct {
-	DNS       DNS        `json:"dns"`
-	Network   Network    `json:"network"`
-	Node      Node       `json:"node"`
-	ApiServer ApiServer  `json:"apiServer"`
-	Debugging Debugging  `json:"debugging"`
-	Etcd      EtcdConfig `json:"etcd"`
+	DNS       DNS       `json:"dns"`
+	Network   Network   `json:"network"`
+	Node      Node      `json:"node"`
+	ApiServer ApiServer `json:"apiServer"`
+	Debugging Debugging `json:"debugging"`
+	Etcd      Etcd      `json:"etcd"`
+}
+
+const (
+	// Etcd performance degrades significantly if the memory available is less than 50MB, enfore this minimum.
+	EtcdMinimumMemoryLimit = 50.0
+)
+
+type Etcd struct {
+	// Set a memory limit, in megabytes, on the etcd process; etcd will begin paging memory when it gets to this value. 0 means no limit.
+	MemoryLimit uint64 `json:"memoryLimit"`
```

Comment:
We have units in some of the other names, even though they aren't exposed. What do you think about including units in this exposed name?

```suggestion
	MemoryLimitMB uint64 `json:"memoryLimitMB"`
```